libc-test/build.rs

```diff
@@ -289,6 +298,24 @@ fn main() {
             "FILE_ATTRIBUTE_INTEGRITY_STREAM" |
             "ERROR_NOTHING_TO_TERMINATE" if mingw => true,
 
+            // these are not in musl
+            "SOL_NETLINK" |
+            "SOL_UDP" |
+            "SOL_AX25" |
+            "SOL_ATALK" |
+            "SOL_NETROM" |
+            "SOL_IPX" |
+            "SOL_ROSE" if musl => true,
+
+            // these are not in glibc or musl yet (but they are in android)
+            "SOL_SCTP" |
+            "SOL_UDPLITE" |
+            "SOL_NETBEUI" |
+            "SOL_LLC" |
+            "SOL_DCCP" |
+            "SOL_NETLINK" |
+            "SOL_TIPC" if linux => true,
```

Comment:
Working with glibc to verify these seems like it might be a reasonable approach although they probably move forward at a snails pace for fear of breaking existing code that has already defined constants missing from its headers (I found several major projects defining `SOL_NETLINK` themselves, some without an `#ifndef` check).  That is, I think the lack of definition of these constants might have less to do with stability and more to do with the inability to have nice things when working in C.

Another angle would be to verify some definitions against the kernel headers.  I don't see a good way of mixing glibc headers and kernel headers together, but having a separate set of tests which verify against kernel sources might have some legs.